src/ChipInput.js

```diff
@@ -203,7 +207,21 @@ class ChipInput extends React.Component {
     if (this.state.focusedChip != null) {
       this.setState({ focusedChip: null })
     }
-    if (this.props.clearOnBlur) {
+    if (this.props.blurBehavior === 'add') {
+      // Lets assume that we only want to add the existing content as chip, when
+      // another event has not added a chip within 200ms .
+      // e.g. onSelection Callback in Autocomplete case
+      let numChipsBefore = (this.props.value || this.state.chips).length
+      let value = event.target.value
+      this.inputBlurTimeout = setTimeout(() => {
```

Comment:
Hi @quorak - wanted to see how an `onSelection` event doesn't complete before `onBlur`? Does `onSelection` blur the input?

I've removed the `setTimeout` in my branch and removed the test that checks if `setTimeout` was called, and nothing else fails. It seems there is no test to illustrate how `onSelection` fails if `setTimeout` is not used.

Ideally, there is a callback option that `onSelection` and `onBlur` can both hook into (maybe a  callback pipeline of some sort (if that is even necessary). However, an alternative is to simply add another props option to toggle `setTimeout` behavior.

/cc @leMaik

Reply:
@quorak I opted for adding the `delayBeforeAdd` option that does not use `setTimeout` by default in #263

Reply:
> It seems there is no test to illustrate how onSelection fails if setTimeout is not used.

@joelvh A new test for this would be :100: so that we have one example for a case where this is needed.

Reply:
@leMaik agreed - I just don't know exactly what @quorak's use case was.